src/modules/MessageSearch/context/__test__/useMessageSearch.spec.tsx

```diff
@@ -0,0 +1,131 @@
+import { MessageSearchProvider } from '../../context/MessageSearchProvider';
+import useMessageSearch from '../../context/hooks/useMessageSearch';
+import { GroupChannel } from '@sendbird/chat/groupChannel';
+import { MessageSearchQuery } from '@sendbird/chat/message';
+
+jest.mock('../../../../hooks/useSendbirdStateContext', () => ({
+  __esModule: true,
+  default: jest.fn(() => ({
+    stores: {
+      sdkStore: {
+        sdk: {
+          createMessageSearchQuery: jest.fn(() => ({
+            next: jest.fn().mockResolvedValue([{ messageId: 1 }]),
+          })),
+        },
+        initialized: true,
+      },
+    },
+    config: { logger: console },
+  })),
+}));
+
+describe('useMessageSearch', () => {
+  const wrapper = ({ children }) => (
+    <MessageSearchProvider channelUrl="test-channel">
+      {children}
+    </MessageSearchProvider>
+  );
+
+  it('provides initial state', () => {
+    const { result } = renderHook(() => useMessageSearch(), { wrapper });
+
+    expect(result.current.state).toEqual(expect.objectContaining({
+      allMessages: [],
+      loading: false,
+      isInvalid: false,
+      initialized: false,
+      currentChannel: null,
+      currentMessageSearchQuery: null,
+      hasMoreResult: false,
+    }));
+  });
+
+  it('updates state when setCurrentChannel is called', () => {
+    const { result } = renderHook(() => useMessageSearch(), { wrapper });
+
+    act(() => {
+      result.current.actions.setCurrentChannel({ url: 'test-channel' } as GroupChannel);
+    });
+
+    expect(result.current.state.currentChannel).toEqual({ url: 'test-channel' });
+    expect(result.current.state.initialized).toBe(true);
+  });
+
+  it('updates state when startMessageSearch is called', () => {
+    const { result } = renderHook(() => useMessageSearch(), { wrapper });
+
+    act(() => {
+      result.current.actions.startMessageSearch();
+    });
+
+    expect(result.current.state.isInvalid).toBe(false);
+    expect(result.current.state.loading).toBe(false);
+  });
+
+  it('updates state when getSearchedMessages is called', () => {
+    const { result } = renderHook(() => useMessageSearch(), { wrapper });
+    const messages = [{ messageId: 1 }, { messageId: 2 }];
+    const mockQuery = { channelUrl: 'test-channel', hasNext: true } as MessageSearchQuery;
+
+    act(() => {
+      result.current.actions.startGettingSearchedMessages(mockQuery);
+      result.current.actions.getSearchedMessages(messages as any, mockQuery);
+    });
+
+    expect(result.current.state.allMessages).toEqual(messages);
+    expect(result.current.state.loading).toBe(false);
+    expect(result.current.state.hasMoreResult).toBe(true);
+  });
+
+  it('updates state when setQueryInvalid is called', () => {
+    const { result } = renderHook(() => useMessageSearch(), { wrapper });
+
+    act(() => {
+      result.current.actions.setQueryInvalid();
+    });
+
+    expect(result.current.state.isInvalid).toBe(true);
```

Comment:
While I was reading the code and the test, it was a bit confusing for me what `isInvalid` means because both a channel and a query could go invalid. Do you mind changing the name to `isQueryInvalid` for clarity?

Reply:
Yeah I got your point and don't mind at all. Updated the prop name in here https://github.com/sendbird/sendbird-uikit-react/pull/1228/commits/35702bac83c4f072e4cbcc6faed8f6d6211e2ed9